imagepruner/v1/types.go

```diff
@@ -0,0 +1,97 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConfigList is a slice of Config objects.
+type ConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []Config `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Config is the configuration object for an image registry pruner
+// managed by the registry operator.
+type Config struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec ImagePrunerSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// +optional
+	Status ImagePrunerStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+}
+
+// PrunerHistory is a configuration object for how many
+// successful and failed finished jobs to retain.
+type PrunerHistory struct {
+	// SuccessfulJobsHistoryLimit specifies how many successful finished jobs to retain.
+	// Defaults to 3 if not set
+	// +optional
+	SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit" protobuf:"bytes,1,opt,name=successfulJobsHistoryLimit"`
+	// FailedJobsHistoryLimit specifies how many failed finished jobs to retain.
+	// Defaults to 3 if not set
+	// +optional
+	FailedJobsHistoryLimit *int32 `json:"failedJobsHistoryLimit" protobuf:"bytes,2,opt,name=failedJobsHistoryLimit"`
+}
+
+type ImagePrunerSpec struct {
+	// Schedule specifies when to execute the job
+	// Uses standard cronjob syntax: https://wikipedia.org/wiki/Cron
+	// Defaults to 0 0 * * *
+	// +optional
+	Schedule string `json:"schedule" protobuf:"bytes,1,opt,name=schedule"`
+	// Suspend specifies wether or not to suspend subsequent executions of this cronjob.
+	// Defaults to false
+	// +required
+	Suspend *bool `json:"suspend" protobuf:"bytes,2,opt,name=suspend"`
```

Comment:
If this is required, should not be a pointer. You also don't need the `// +required` tag.
If the underlying `CronJob` accepts a pointer, then let's make this optional.

Reply:
updated in next push